use/wordcount-example.md

```diff
@@ -0,0 +1,413 @@
+
+In this example, two transforms are encapsulated as the `PTransform` subclass `CountWords`. `CountWords` contains the `ParDo` that runs `ExtractWordsFn` and the SDK-provided `Count` transform.
+
+When `CountWords` is defined, we specify its ultimate input and output; the input is the `PCollection<String>` for the extraction operation, and the output is the `PCollection<KV<String, Long>>` produced by the count operation.
+
+```
+public static class CountWords extends PTransform<PCollection<String>,
+    PCollection<KV<String, Long>>> {
+  @Override
+  public PCollection<KV<String, Long>> apply(PCollection<String> lines) {
+
+    // Convert lines of text into individual words.
+    PCollection<String> words = lines.apply(
+        ParDo.of(new ExtractWordsFn()));
+
+    // Count the number of times each word occurs.
+    PCollection<KV<String, Long>> wordCounts =
+        words.apply(Count.<String>perElement());
+
+    return wordCounts;
+  }
+}
+
+public static void main(String[] args) throws IOException {
+  Pipeline p = ...
+
+  p.apply(...)
+   .apply(new CountWords())
+   ...
+}
+```
+
+### Using Parameterizable PipelineOptions
+
+You can hard-code various execution options when you run your pipeline. However, the more common way is to define your own configuration options via command-line argument parsing. 
+
+Add arguments to be processed by the command-line parser, and specify default values for them. You can then access the options values in your pipeline code.
+
+```
+public static interface WordCountOptions extends PipelineOptions {
+  @Description("Path of the file to read from")
+  @Default.String("gs://dataflow-samples/shakespeare/kinglear.txt")
+  String getInputFile();
+  void setInputFile(String value);
+  ...
+}
+
+public static void main(String[] args) {
+  WordCountOptions options = PipelineOptionsFactory.fromArgs(args).withValidation()
+      .as(WordCountOptions.class);
+  Pipeline p = Pipeline.create(options);
+  ...
+}
+```
+
+## Debugging WordCount Example
+
+The Debugging WordCount example demonstrates some best practices for instrumenting your pipeline code.
+
+To run this example, follow the instructions in the [Beam Examples README](https://github.com/apache/incubator-beam/blob/master/examples/java/README.md#building-and-running). To view the full code, see **[DebuggingWordCount](https://github.com/apache/incubator-beam/blob/master/examples/java/src/main/java/org/apache/beam/examples/DebuggingWordCount.java).**
+
+**New Concepts:**
+
+* Logging
+* Testing your Pipeline via PAssert
+
+The following sections explain these key concepts in detail, and break down the pipeline code into smaller sections.
+
+### Logging
+
+Each runner may choose to handle logs in its own way. 
```

Comment:
Add subheadings Direct, Apache Spark, Apache Flink, and Dataflow.  

Direct runner will just print the log messages directly to your local console. @tgroh in case he wants to suggest anything different.
File jira issues to explain how logging works on Flink and Spark.